### PR TITLE
[Hub Generated] Review request for Weather to add version preview/1.0

### PR DESCRIPTION
**Pull request by john35452**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/john35452/azure-rest-api-specs/dev-maps-Weather-1.0...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/maps/data-plane/Weather/preview/1.0/weather.json [hunk 1]
```diff
@@ -506,6 +514,155 @@
...
+          {
+            "name": "query",
+            "description": "Coordinates of the location for which air quality is requested. The applicable query is specified as a comma separated string composed by latitude followed by longitude e.g. \"47.641268,-122.125679\".",
+            "in": "query",
+            "type": "string",
+            "required": true
+          },
+          {
+            "name": "duration",
+            "description": "Specifies for how many hours from now we would like to know about the air quality. Available values are 1, 12, 24, 48, 72, 96.",
```
Comment (sataneja) [inline specification/maps/data-plane/Weather/preview/1.0/weather.json:645]: Just like in Daily forcast we shd add the default value here as well.

File: specification/maps/data-plane/Weather/preview/1.0/weather.json [hunk 2]
```diff
@@ -1910,6 +2067,255 @@
...
+          "description": "A list of all daily air quality forecasts for the queried location."
+        }
+      }
+    },
+    "AirQuality": {
+      "type": "object",
+      "description": "Information about the air quality in a specific location at a specific time.",
+      "properties": {
+        "dateTime": {
+          "type": "string",
```
Comment (ambientlight) [inline specification/maps/data-plane/Weather/preview/1.0/weather.json:2115]: `"format": "date-time"` will be a good addition here as the proceeding codegen will result in date types rather then raw strings:  https://github.com/ambientlight/azure-sdk-for-go/blob/825aa43364c936dfde70094d465ab8e7e07eef10/services/preview/maps/1.0/weather/models.go#L45

File: specification/maps/data-plane/Weather/preview/1.0/weather.json [hunk 3]
```diff
@@ -1910,6 +2067,255 @@
...
+        }
+      }
+    },
+    "DailyAirQuality": {
+      "type": "object",
+      "description": "Information about the air quality in a specific location at a specific time.",
+      "properties": {
+        "dateTime": {
+          "type": "string",
+          "description": "Date and time of the current observation displayed in ISO 8601 format, for example, 2019-10-27T19:39:57-08:00."
```
Comment (ambientlight) [inline specification/maps/data-plane/Weather/preview/1.0/weather.json:2159]: `"format": "date-time"` will be a good addition here as the proceeding codegen will result in date types rather then raw strings:  https://github.com/ambientlight/azure-sdk-for-go/blob/825aa43364c936dfde70094d465ab8e7e07eef10/services/preview/maps/1.0/weather/models.go#L313

**Review by ambientlight — COMMENTED** [review]
I tested the codegen on the derived swagger updates and it is good on my side besides the mentioned `"format": "date-time"` update

File: specification/maps/data-plane/Weather/preview/1.0/weather.json (hunk 1 above)
Comment (john35452) [reply specification/maps/data-plane/Weather/preview/1.0/weather.json]: Updated

File: specification/maps/data-plane/Weather/preview/1.0/weather.json (hunk 2 above)
Comment (john35452) [reply specification/maps/data-plane/Weather/preview/1.0/weather.json]: Updated

File: specification/maps/data-plane/Weather/preview/1.0/weather.json (hunk 3 above)
Comment (john35452) [reply specification/maps/data-plane/Weather/preview/1.0/weather.json:2159]: Updated

**Review by evaou — APPROVED** [review]

File: specification/maps/data-plane/Weather/preview/1.0/weather.json [hunk 4]
```diff
@@ -506,6 +507,155 @@
           }
         }
       }
+    },
+    "/weather/airQuality/current/{format}": {
+      "get": {
+        "description": "**Get Current Air Quality**\n\n**Applies to**: S0 and S1 pricing tiers.\n\nThe service provides the detailed information about the concentration of pollutant and overall status for current air quality.",
```
Comment (evaou) [inline specification/maps/data-plane/Weather/preview/1.0/weather.json:513]: in code, SKU is S1 and G2.

**Pull request closed without merging**